numba/tests/test_typedlist.py

```diff
@@ -729,6 +729,33 @@ def impl():
         got = impl()
         self.assertEqual(expected, got)
 
+    def test_extend_single_value_container(self):
+        @njit
+        def impl():
+            l = List()
+            l.extend((100,))
+            return l
+
+        expected = impl.py_func()
+        got = impl()
+        self.assertEqual(expected, got)
+
+    def test_extend_empty_unrefined(self):
+        # Extending an unrefined list with an empty iterable doesn't work in a
+        # jit compiled function as the list remains untyped.
+        l = List()
+        l.extend(tuple())
+        self.assertEqual(len(l), 0)
```

Comment:
Is this behaviour a good idea? What about JIT transparency etc.

Reply:
Also, if you put this in a JIT compiled function, you get:
```
Invalid use of Function(<function impl_extend at 0x7f5648715c10>) with argument(s) of type(s): (ListType[undefined], Tuple())
 * parameterized
In definition 0:
    TypingError: extend argument must be iterable
```
empty tuples are iterable, this is a bit confusing. I think it's arising because an empty tuple is typed as `Tuple` at present, which is not iterable for Numba's purposes.

Reply:
Yeah, I think only `UniTuple` is iterable. I struggled with this one for some time and came to the conclusion that this is a corner case that only works outside of a `jit` context.

Reply:
So, even if `tuple()` would be something iterable, the function would not compile as the list can not be refined.